pkg/network/state.go

```diff
@@ -734,20 +710,23 @@ func (ns *networkState) determineConnectionIntraHost(connections []ConnectionSta
 	}
 }
 
-func addConnections(a, b *ConnectionStats) {
-	a.Monotonic.SentBytes += b.Monotonic.SentBytes
-	a.Monotonic.RecvBytes += b.Monotonic.RecvBytes
-	a.Monotonic.SentPackets += b.Monotonic.SentPackets
-	a.Monotonic.RecvPackets += b.Monotonic.RecvPackets
-	a.Monotonic.Retransmits += b.Monotonic.Retransmits
-	a.Monotonic.TCPEstablished += b.Monotonic.TCPEstablished
-	a.Monotonic.TCPClosed += b.Monotonic.TCPClosed
+func mergeConnectionStats(a, b *ConnectionStats) {
+	for cookie, bm := range b.Monotonic {
+		if am, ok := a.Monotonic[cookie]; ok {
+			a.Monotonic[cookie] = am.Max(bm)
+			continue
+		}
+
+		a.Monotonic[cookie] = bm
+	}
 
 	if b.LastUpdateEpoch > a.LastUpdateEpoch {
 		a.LastUpdateEpoch = b.LastUpdateEpoch
 	}
 
-	if a.IPTranslation == nil {
-		a.IPTranslation = b.IPTranslation
+	if (a.IPTranslation != nil && b.IPTranslation != nil && *a.IPTranslation != *b.IPTranslation) ||
+		(a.IPTranslation != nil && b.IPTranslation == nil) ||
+		(a.IPTranslation == nil && b.IPTranslation != nil) {
+		a.IPTranslation = nil
```

Comment:
Curious about why we would set `a.IPTranslation = nil`

Reply:
This is trying to resolve a "conflict" in the nat info for the same key. Only if both connections have NAT info that matches or don't have NAT info do we keep the NAT info for connection `a` which is being updated. There is really no way to determine which NAT info is correct in cases where we have two different NAT infos or one connection has NAT info and the other doesn't.